storage/src/main/java/org/apache/kafka/storage/internals/log/RemoteIndexCache.java

```diff
@@ -451,7 +456,7 @@ public static class Entry implements AutoCloseable {
 
         private final OffsetIndex offsetIndex;
         private final TimeIndex timeIndex;
-        private final TransactionIndex txnIndex;
+        private final Optional<TransactionIndex> txnIndexOpt;
```

Comment:
We should make it clear again why we treat txnIndex differently. Please add a comment. Thanks.

Reply:
Added